monai/utils/type_conversion.py

```diff
@@ -366,3 +385,72 @@ def convert_to_list(data: Union[Sequence, torch.Tensor, np.ndarray]) -> list:
 
     """
     return data.tolist() if isinstance(data, (torch.Tensor, np.ndarray)) else list(data)
+
+
+def get_dtype_bound_value(dtype: Union[DtypeLike, torch.dtype]):
+    """
+    Get dtype bound value
+    Args:
+        dtype: dtype to get bound value
+    Returns:
+        (bound_min_value, bound_max_value)
+    """
+    if dtype in UNSUPPORTED_TYPES:
+        is_floating_point = False
+    else:
+        is_floating_point = get_equivalent_dtype(dtype, torch.Tensor).is_floating_point
+    dtype = get_equivalent_dtype(dtype, np.array)
+    if is_floating_point:
+        return (np.finfo(dtype).min, np.finfo(dtype).max)  # type: ignore
+    else:
+        return (np.iinfo(dtype).min, np.iinfo(dtype).max)
+
+
+def safe_dtype_convert(data: Any, dtype: Union[DtypeLike, torch.dtype] = None):
```

Comment:
this introduces some duplicated logic with the existing code. could this function simply do the data value change but not change the dtype?

Reply:
Hi @wyli, docstring is not quite right, this function actually only does data value change, not change dtype and type.

Reply:
I see, perhaps change the function name to `safe_dtype_range` as well and please write a new test class for this function as it might be used by many components in the future. for example that np.clip might not work with cupy arrays.

Reply:
Hi @wyli, I have added new unit tests and found that when python version > 3.10, the value cannot be converted to type uint8 without overflow which caused CI to fail.
https://github.com/Project-MONAI/MONAI/actions/runs/3626898100/jobs/6116284269

Not sure about this case, maybe In addition to adding the safe flag as I am doing now, we should first give a warning if it overflows but still can't solve the error unless forced safe convert. Do you have any suggestions? Thanks!

Reply:
which specific case is this?

Reply:
raising an runtime error when safe=False is actually a better handling, we can add the same exception for the unsafe cases or treat the runtime error as expected.